Init redux

skipGit opened the pull request:
1. Install redux, react-redux.
2. Create and configure store.
3. Provide store to App.

skipGit committed:
init-redux

1. Install redux (reduxjs/toolkit, react-redux).
2. Create Redux Store and provide it to App.
3. Create a Redux State Slice and add Slice Reducer to the Store.

skipGit committed:
init-redux

1. Install redux, react-redux.
2. Create and configure store.
3. Provide store  to App.

skipGit committed:
init-redux

Remove unnecessary redux-toolkit associated files.

DmitryKirilenko16 commented on src/store/actionCreators/actions.js:
```diff
@@ -0,0 +1,20 @@
+  TOGGLE_TODO,
+} from "../actionsTypes/actionTypes";
+
+export const addTodo = (message) => ({
+  type: ADD_TODO,
+  payload: { message },
+});
+
+export const toggleTodo = (id, isCompleted) => ({
+  type: TOGGLE_TODO,
+  payload: { id, isCompleted },
+});
+
+export const deleteTodo = (id) => ({
+  type: DELETE_TODO,
+  payload: { id },
+});
```
Create simpleActionGenerator func that will return a new such object {payload: payloadFromParam, type: typeFromParam}

DmitryKirilenko16 replied:
Will be a generic action creator:
const actionCreator1 = (payload) => simpleActionGenerator(TYPE, payload)

DmitryKirilenko16 commented on src/store/reducers/todos.js:
```diff
@@ -0,0 +1,28 @@
+        message: action.payload.message,
+        isCompleted: false,
+      };
+      state.push(todo);
```
Reducer is a pure function. You should not mutate the state calling push(), pop(), etc
You should create a new store copy on every change

skipGit committed:
init-redux

1. Configured store.
2. Described reducers.
3. Described constants and actions.

DmitryKirilenko16 commented on src/store/toDo/reducer.js:
```diff
@@ -0,0 +1,31 @@
+    case DELETE_TODO:
+      return {
+        ...state,
+        todos: state.todos.filter((todo) => todo.id !== action.payload),
```
should be action.payload.todoId

DmitryKirilenko16 commented on src/store/index.js:
```diff
@@ -0,0 +1,5 @@
+import { composeWithDevTools } from "redux-devtools-extension";
+import { toDoReducer } from "./toDo/reducer";
+
+export const store = createStore(toDoReducer, composeWithDevTools());
```
create a root reducer and pass toDoReducer here

skipGit committed:
init-redux

1. Added simpleActiongenerator.
2. Added rootReducer and pass toDoReducer in it.
